masu/util/common.py

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Common util functions."""
+
+import re
+
+
+def extract_uuids_from_string(source_string):
+    """
+    Extract uuids out of a given source string.
+
+    Args:
+        source_string (Source): string to locate UUIDs.
+
+    Returns:
+        ([]) List of UUIDs found in the source string
+
+    """
+    uuid_regex = '[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}'
+    found_uuid = re.findall(uuid_regex, source_string.lower())
```

Comment:
This is a form of yoda condition. 

It's changing the input to match the regex, rather than constructing the regex to match the input (either with `re.IGNORECASE` or by using `[0-9a-fA-F]`).

They're functionally identical, but the preferred style is to hold the input constant and match it with a well-constructed regex.